### PR TITLE
Organize Design Proposals Dir by SIG

### LINKED ISSUE #918: Organize Design Proposals Folder
The design proposals folder is getting a little out of hand, and we could do a few things to help improve it.

- [ ] In lieu of sig subdirs, create functional area subdirs with clear ownership
- [ ] Create OWNERS files in those subdirs
- [ ] Disambiguate proposals from accepted changes as tracked by #914

### PR DESCRIPTION
**Pull request by grodrigues3**
Organize design proposals directory into sigs.  This will make it easier to assign reviewers for future proposals.

Fixes #918

### CODE REVIEW

**Comment by liggitt** [comment]
What's the plan for proposals involving multiple sigs?

**Comment by grodrigues3** [comment]
I'm open to suggestions.  a **multi-sig** directory or something equivalent?  maybe just the base dir?

**Comment by bgrant0607** [comment]
@liggitt I haven't reviewed this yet, but my thinking is as follows:

* There should usually be a primary SIG owner, but that doesn't preclude notifying other SIGs and adding approvers from other SIGs
* There are many SIGs with project-wide impact, which would normally be the primary owners
* Proposals/designs/conventions that truly span the whole project likely would go into sig-architecture
* There are some things in design-proposals that probably don't belong there

See discussion in #967 and https://groups.google.com/forum/#!topic/kubernetes-sig-architecture/t-1EqeEoLPA

cc @kubernetes/sig-architecture-pr-reviews @smarterclayton

**Comment by bgrant0607** [comment]
@grodrigues3 Were there any you weren't sure about? 

Why are some files deleted and added rather than just moved?

This isn't as easy to read as a doc or spreadsheet. The github UI abbreviates the paths.

**Comment by bgrant0607** [comment]
@grodrigues3 A number of these will need to be changed. Could we please start with a doc, spreadsheet, or something easier to read and edit or comment on?

**Comment by smarterclayton** [comment]
Also note that you deleted some proposals :)

**Comment by grodrigues3** [comment]
Yeah, I'll start with a sheet.

If I deleted any, it was a mistake.

I created folders for sigs, and just dragged proposals over.

**Comment by grodrigues3** [comment]
How's [this](https://github.com/kubernetes/community/pull/1010/files#diff-61ab3f0d5e63b88ac095bbb0bb5154de), can you comment on that and I change move and update things accordingly?

File: contributors/design-proposals/dir_struct.txt [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+	downward_api_resources_limits_requests.md
+	seccomp.md
+	client-package-structure.md
+	service-discovery.md
+	metadata-policy.md
+	containerized-mounter.md~
+	identifiers.md
+	local-cluster-ux.md
+	pod-pid-namespace.md
+	grow-volume-size.md
```
Comment (msau42) [inline contributors/design-proposals/dir_struct.txt]: sig storage

**Comment by bgrant0607** [comment]
Reminder: github doesn't send notifications when commits are pushed.

File: contributors/design-proposals/dir_struct.txt [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+	image-provenance.md
+	core-metrics-pipeline.md
+	versioning.md
+	ha_master.md
+	secret-configmap-downwarapi-file-mode.md
+	protobuf.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: API machinery

File: contributors/design-proposals/dir_struct.txt [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+	flakiness-sla.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Testing

File: contributors/design-proposals/dir_struct.txt [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+	resources.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Scheduling, or Resource Management WG if we want to allow WGs to own anything

File: contributors/design-proposals/dir_struct.txt [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+	initial-resources.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Autoscaling

File: contributors/design-proposals/dir_struct.txt [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+	create_sheet.py
+	runtime-client-server.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node, but worth asking if it's obsolete

File: contributors/design-proposals/dir_struct.txt [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+	OWNERS
+	namespaces.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Architecture

File: contributors/design-proposals/dir_struct.txt [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+	cpu-manager.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: It's pretty inconvenient that these aren't alphabetical.

Node, or Resource Management WG

File: contributors/design-proposals/dir_struct.txt [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+	selinux-enhancements.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+	sysctl.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+	dir_struct.txt
+	selinux.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+	templates.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: We should mark this as obsolete, but I want to preserve the history.

Apps

File: contributors/design-proposals/dir_struct.txt [hunk 13]
```diff
@@ -0,0 +1,240 @@
...
+	pod-cache.png
+	README.md
+	multi-platform.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: We need to decide who owns the build system. Obvious candidates are Release, Testing, Contributor Experience, and new SIG.

File: contributors/design-proposals/dir_struct.txt [hunk 14]
```diff
@@ -0,0 +1,240 @@
...
+	pod-lifecycle-event-generator.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 15]
```diff
@@ -0,0 +1,240 @@
...
+	secrets.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Auth

File: contributors/design-proposals/dir_struct.txt [hunk 16]
```diff
@@ -0,0 +1,240 @@
...
+	cri-dockershim-checkpoint.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 17]
```diff
@@ -0,0 +1,240 @@
...
+	event_compression.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: We need to decide who owns Events. I'm tempted to say API machinery.

File: contributors/design-proposals/dir_struct.txt [hunk 18]
```diff
@@ -0,0 +1,240 @@
...
+	pleg.png
+./sig-cli
+	kubectl-extension.md
+	kubectl_apply_getsetdiff_last_applied_config.md
+	multi-fields-merge-key.md
+	template.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Is this different from templates.md?

File: contributors/design-proposals/dir_struct.txt [hunk 19]
```diff
@@ -0,0 +1,240 @@
...
+	expansion.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: This is done in Kubelet. Node.

File: contributors/design-proposals/dir_struct.txt [hunk 20]
```diff
@@ -0,0 +1,240 @@
...
+	device-plugin.png
+	gpu-support.md
+	device-plugin-overview.png
+./autoscaling
+	hpa-v2.md
+	hpa-status-conditions.md
+	horizontal-pod-autoscaler.md
+./architecture
+	architecture.md
+	architecture.dia
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: We should just delete these old images.

File: contributors/design-proposals/dir_struct.txt [hunk 21]
```diff
@@ -0,0 +1,240 @@
...
+	architecture.png
+	architecture.svg
+./api-machinery
+	admission_control_extension.md
+	csi-client-structure-proposal.md
+	selector-generation.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Apps

File: contributors/design-proposals/dir_struct.txt [hunk 22]
```diff
@@ -0,0 +1,240 @@
...
+	pod-safety.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: I'd say storage, though it's also relevant to node and apps.

File: contributors/design-proposals/dir_struct.txt [hunk 23]
```diff
@@ -0,0 +1,240 @@
...
+	container-init.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 24]
```diff
@@ -0,0 +1,240 @@
...
+	resource-quota-scoping.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Resource management WG

File: contributors/design-proposals/dir_struct.txt [hunk 25]
```diff
@@ -0,0 +1,240 @@
...
+	thirdpartyresources.md
+	aggregated-api-servers.md
+	extending-api.md
+	envvar-configmap.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 26]
```diff
@@ -0,0 +1,240 @@
...
+	dynamic-admission-control-configuration.md
+	api-chunking.md
+	garbage-collection.md
+	customresources-validation.md
+	auditing.md
+	apiserver-watch.md
+	admission_control_limit_range.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Resource management WG

File: contributors/design-proposals/dir_struct.txt [hunk 27]
```diff
@@ -0,0 +1,240 @@
...
+	apiserver-build-in-admission-plugins.md
+	synchronous-garbage-collection.md
+	configmap.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 28]
```diff
@@ -0,0 +1,240 @@
...
+	csi-new-client-library-procedure.md
+	pod-preset.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Service catalog

File: contributors/design-proposals/dir_struct.txt [hunk 29]
```diff
@@ -0,0 +1,240 @@
...
+	add-new-patchStrategy-to-clear-fields-not-present-in-patch.md
+	api-group.md
+	principles.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Architecture

File: contributors/design-proposals/dir_struct.txt [hunk 30]
```diff
@@ -0,0 +1,240 @@
...
+	admission_control.md
+	optional-configmap.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 31]
```diff
@@ -0,0 +1,240 @@
...
+	server-get.md
+	apiserver-count-fix.md
+	admission_control_resource_quota.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Resource management WG

File: contributors/design-proposals/dir_struct.txt [hunk 32]
```diff
@@ -0,0 +1,240 @@
...
+./node
+	pod-resource-management.md
+	kubelet-tls-bootstrap.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Cluster lifecycle

File: contributors/design-proposals/dir_struct.txt [hunk 33]
```diff
@@ -0,0 +1,240 @@
...
+	dynamic-kubelet-configuration.md
+	kubelet-hypercontainer-runtime.md
+	container-runtime-interface-v1.md
+	kubelet-authorizer.md
+	disk-accounting.md
+	kubelet-systemd.md
+	kubelet-cri-logging.md
+	kubelet-auth.md
+	runtimeconfig.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Cluster lifecycle

File: contributors/design-proposals/dir_struct.txt [hunk 34]
```diff
@@ -0,0 +1,240 @@
...
+	volume_stats_pvc_ref.md
+./auth
+	security_context.md
+	no-new-privs.md
+	access.md
+	enhance-pluggable-policy.md
+	apparmor.md
+	security-context-constraints.md
+	pod-security-context.md
+	bulk_watch.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: API machinery

File: contributors/design-proposals/dir_struct.txt [hunk 35]
```diff
@@ -0,0 +1,240 @@
...
+	dynamic.png
+./release
+	release-notes.md
+	release-test-signal.md
+./scheduling
+	rescheduling.md
+	rescheduler.md
+	nodeaffinity.md
+	podaffinity.md
+	hugepages.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Resource management WG

File: contributors/design-proposals/dir_struct.txt [hunk 36]
```diff
@@ -0,0 +1,240 @@
...
+	pod-preemption.md
+	pod-priority-api.md
+	taint-node-by-condition.md
+	scheduler_extender.md
+	rescheduling-for-critical-pods.md
+	multiple-schedulers.md
+./apps
+	daemonset-update.md
+	cronjob.md
+	annotations-downward-api.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Downward API is Node

File: contributors/design-proposals/dir_struct.txt [hunk 37]
```diff
@@ -0,0 +1,240 @@
...
+	controller-ref.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: API machinery

File: contributors/design-proposals/dir_struct.txt [hunk 38]
```diff
@@ -0,0 +1,240 @@
...
+	statefulset-update.md
+	stateful-apps.md
+	deploy.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Also relevant to CLI, but Apps is fine

File: contributors/design-proposals/dir_struct.txt [hunk 39]
```diff
@@ -0,0 +1,240 @@
...
+	daemon.md
+	controller_history.md
+	job.md
+	indexed-job.md
+	deployment.md
+./storage
+	flex-volumes-drivers-psp.md
+	local-storage-overview.md
+	all-in-one-volume.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: Node, actually

File: contributors/design-proposals/dir_struct.txt [hunk 40]
```diff
@@ -0,0 +1,240 @@
...
+	volume-selectors.md
+	persistent-storage.md
+	volume-metrics.md
+	flexvolume-deployment.md
+	volume-snapshotting.png
+	volume-provisioning.md
+	propagation.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: I'd put this under node

File: contributors/design-proposals/dir_struct.txt [hunk 41]
```diff
@@ -0,0 +1,240 @@
...
+	volume-ownership-management.md
+	mount-options.md
+	volumes.md
+	default-storage-class.md
+	volume-snapshotting.md
+	volume-hostpath-qualifiers.md
+./aws
+	aws_under_the_hood.md
+./images
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: These are for the scheduling preemption proposal, so scheduling

File: contributors/design-proposals/dir_struct.txt [hunk 42]
```diff
@@ -0,0 +1,240 @@
...
+	preemption_1.png
+	preemption_3.png
+	.gitignore
+	OWNERS
+	preemption_2.png
+	preemption_4.png
+./gcp
+	gce-l4-loadbalancer-healthcheck.md
+	containerized-mounter.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: This is probably Storage. I don't think it's GCP-specific.

**Comment by bgrant0607** [comment]
I didn't have much problem categorizing everything by SIG. I think that approach will work fine. We'll need a way to notify other SIGs that may be interested.

**Comment by bgrant0607** [comment]
Wearing my steering-committee hat, eventually I want a bidirectional mapping from SIG to code (in SIG charter) and code to SIG (in OWNERS).

File: contributors/design-proposals/dir_struct.txt (hunk 18 above)
Comment (grodrigues3) [reply contributors/design-proposals/dir_struct.txt]: My mistake.  This is the design proposal template

File: contributors/design-proposals/dir_struct.txt (hunk 20 above)
Comment (grodrigues3) [reply contributors/design-proposals/dir_struct.txt]: are they all outdated or just the dia?

File: contributors/design-proposals/dir_struct.txt (hunk 6 above)
Comment (grodrigues3) [reply contributors/design-proposals/dir_struct.txt]: @dchen1107 @derekwaynecarr can we mark runtime-client-server.md as obsolete?

File: contributors/design-proposals/dir_struct.txt (hunk 20 above)
Comment (bgrant0607) [reply contributors/design-proposals/dir_struct.txt]: dia, png, svg

**Comment by bgrant0607** [comment]
Thanks @grodrigues3. Maybe we could find someone else from @kubernetes/sig-architecture-pr-reviews to proofread the changes?

**Comment by grodrigues3** [comment]
@bgrant0607 looks like you're the only one on that team right now.  Can you suggest another reviewer by gh handle?

@smarterclayton perhaps?

File: contributors/design-proposals/dir_struct.txt [hunk 43]
```diff
@@ -0,0 +1,244 @@
+Uncategorized
+	admission_control_event_rate_limit.md
+	create_sheet.py
+	create_sheet.py~
+	design_proposal_template.md
+	dir_struct.txt
+	event_compression.md
```
Comment (smarterclayton) [inline contributors/design-proposals/dir_struct.txt]: Api machinery

File: contributors/design-proposals/dir_struct.txt [hunk 44]
```diff
@@ -0,0 +1,244 @@
+Uncategorized
+	admission_control_event_rate_limit.md
```
Comment (smarterclayton) [inline contributors/design-proposals/dir_struct.txt]: Api machinery

File: contributors/design-proposals/dir_struct.txt [hunk 45]
```diff
@@ -0,0 +1,244 @@
...
+	create_sheet.py
+	create_sheet.py~
+	design_proposal_template.md
+	dir_struct.txt
+	event_compression.md
+	multi-platform.md
+	owners
+	pleg.png
```
Comment (smarterclayton) [inline contributors/design-proposals/dir_struct.txt]: Node

File: contributors/design-proposals/dir_struct.txt [hunk 46]
```diff
@@ -0,0 +1,244 @@
...
+	readme.md
+	runtime-client-server.md
+	templates.md~
```
Comment (smarterclayton) [inline contributors/design-proposals/dir_struct.txt]: Apps

File: contributors/design-proposals/scheduling/hugepages.md [hunk 47]
```diff
@@ -0,0 +1,308 @@
+# HugePages support in Kubernetes
```
Comment (smarterclayton) [inline contributors/design-proposals/scheduling/hugepages.md]: Weird that this didn't get treated as a move

File: contributors/design-proposals/apps/OBSOLETE_templates.md [hunk 48]
```diff
@@ -1,3 +1,5 @@
+# OBSOLETE
```
Comment (smarterclayton) [inline contributors/design-proposals/apps/OBSOLETE_templates.md]: Should probably not be doing this here

Comment (grodrigues3) [reply contributors/design-proposals/apps/OBSOLETE_templates.md]: any suggestions for where/how to mark these deprecated?

**Comment by grodrigues3** [comment]
ping @bgrant0607 @smarterclayton anything else I should change?

File: contributors/design-proposals/apps/OBSOLETE_templates.md (hunk 48 above)
Comment (bgrant0607) [reply contributors/design-proposals/apps/OBSOLETE_templates.md]: Let's wait for the metadata proposal.

https://docs.google.com/document/d/1ynmBMuDuT7yGzRscObB1KtgJj8ljYq0I5q4oshrJUCs/edit#

File: contributors/design-proposals/dir_struct.txt [hunk 49]
```diff
@@ -0,0 +1,242 @@
+Uncategorized
+	create_sheet.py
+	design_proposal_template.md
+	dir_struct.txt
+	multi-platform.md
+	owners
+	readme.md
+	runtime-client-server.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: This should have been Node.

File: contributors/design-proposals/scheduling/hugepages.md (hunk 47 above)
Comment (bgrant0607) [reply contributors/design-proposals/scheduling/hugepages.md]: It was moved to resource-management. This is a dupe.

File: contributors/design-proposals/storage/containerized-mounter.md [hunk 50]
```diff
@@ -2,7 +2,7 @@
 
 ## Goal
 
-Due security and management overhead, our new Container-Optimized OS used by GKE
+Due to security and management overhead, our new Container-Optimized OS used by GKE
```
Comment (bgrant0607) [inline contributors/design-proposals/storage/containerized-mounter.md]: We really shouldn't change anything in this PR

**Comment by bgrant0607** [comment]
This is very close. Thanks @grodrigues3.

File: contributors/design-proposals/dir_struct.txt [hunk 51]
```diff
@@ -0,0 +1,242 @@
...
+	owners
+	preserve-order-in-strategic-merge-patch.md
+	simple-rolling-update.md
+./network
+	command_execution_port_forwarding.md
+	external-lb-source-ip-preservation.md
+	flannel-integration.md
+	network-policy.md
+	networking.md
+	selinux-enhancements.md
```
Comment (bgrant0607) [inline contributors/design-proposals/dir_struct.txt]: This selinux one should be in node.

**Comment by bgrant0607** [comment]
@grodrigues3 
Can we wrap this up by tomorrow?

**Comment by bgrant0607** [comment]
@grodrigues3 What about  contributors/design-proposals/scheduling/hugepages.md ?

**Comment by grodrigues3** [comment]
Removed the dupe hugepages and the OBSOLETE marking at the top of the OBSOLETE_template.md file.

**Comment by bgrant0607** [comment]
Awesome!
/lgtm

**Comment by bgrant0607** [comment]
@grodrigues3 I just noticed that "sig-cli" is the only subdirectory with "sig-" in the name.

**Comment by grodrigues3** [comment]
I was on auto-pilot and didn't update that one.  That was created before my PR.  I'll update and submit another PR

**Comment by php-coder** [comment]
This renaming produced many broken links in the docs :(

**Comment by php-coder** [comment]
> This renaming produced many broken links in the docs :(

@grodrigues3 I fixed some of them but there are more (I used `git grep 'community/contributors/design-proposals/[a-z]\+\.md' to find them`). Should I fill an issue or you will take care of them?

**Comment by bgrant0607** [comment]
@php-coder Could you please send a PR to fix them?

**Comment by php-coder** [comment]
@bgrant0607 Ok, I'll prepare yet another PR.

**Comment by php-coder** [comment]
@grodrigues3 I've created https://github.com/kubernetes/community/pull/1113 but there is still one proposal left (`multi-platform.md`) that should be moved (I don't know in which directory it should be placed).

**Comment by php-coder** [comment]
@bgrant0607 PR has been submitted.

@grodrigues3 Main repository also is affected because it has links to proposals in code comments, fields descriptions and even help messages. JFYI.

**Comment by janetkuo** [comment]
Is there a better way to fix/avoid broken links after GitHub file move/rename?

**Comment by mbohlool** [comment]
This change broke a lot of links in kubernetes-client and kubernetes-incubator/client-python. Among those are badges we have for clients like this one: 
[Client Capabilities]

In previous attempts of moving documents, we keep a move notice in place, why we did not do that here? I think we will have a lot of these broken links to fix.

**Comment by bgrant0607** [comment]
@mbohlool Kubernetes developer-oriented docs should not be pointed at contributor docs.

**Comment by mbohlool** [comment]
I am not sure if I understand your distinction between contributor and
developer. In this specific case that we defined client's capability and
have a badge for it in README files, What do you suggest?

**Comment by bgrant0607** [comment]
@mbohlool 

An example:

Contributors == people sending us PRs (in this context)

This looks like a document intended to be read by developers using the client library to build on top of K8s APIs for their own purposes:
https://github.com/kubernetes-client/python-base/blob/master/README.md

That puts them in a "user" category.

Therefore the docs should be written for users and belong on kubernetes.io or in the client repos themselves.

**Pull request merged**